src/parsers/espresso/init.js

```diff
@@ -0,0 +1,61 @@
+import { APPLICATIONS, NATIVE_FORMATS } from "../enums";
+import BaseParser from "../init";
+import parser7_2 from "./7.2/parser";
+
+const parserMap = {
+    // TODO: change correctly
+    5.2: parser7_2,
+    "5.*": parser7_2,
+    "6.*": parser7_2,
+    7.2: parser7_2,
+};
+
+// eslint-disable-next-line no-unused-vars
+const versionRegexMap = {
+    // TODO: implement version detection
+    5.2: /temp/,
+    "5.*": function () {},
+    "6.*": function () {},
+    "7.*": function () {},
+};
+
+class EspressoParser extends BaseParser {
+    // eslint-disable-next-line no-useless-constructor
+    constructor(content) {
+        super(content);
+    }
+
+    static validate(text) {
+        const version = EspressoParser.getVersionByContent(text);
+        const parser = EspressoParser.getParserByVersion(version);
+        return parser.isEspressoFormat(text);
+    }
+
+    static getVersionByContent(text) {
+        versionRegexMap.forEach((version) => {
+            if (versionRegexMap[version].test(text)) {
+                return version;
+            }
+        });
+        throw new Error("Could not determine version from content.");
+    }
+
+    static getParserByVersion(version) {
+        return parserMap[version];
+    }
+
+    static getIntermediateFormat(text) {
+        const version = EspressoParser.getVersionByContent(text);
+        const parser = EspressoParser.getParserByVersion(version);
+        return {
+            metadata: {
+                application: APPLICATIONS.ESPRESSO,
+                format: NATIVE_FORMATS.QE,
+                version: EspressoParser.getVersionByContent(text),
+            },
+            data: parser.fromEspressoFormat(text),
```

Comment:
Following up on the comment from above (https://github.com/Exabyte-io/made.js/pull/81/files#r1251421507), it would make sense to have the parsing functions as class methods of `EspressoParser`.